### DISCUSSION TITLE
Auth::user()->id problem

### DISCUSSION THREAD

**Discussion by fatihbarissahan** [yajra/laravel-datatables#2808] [General]
Hello,

I tried every way but I can't figure it out. I am using Spatie permission plugin and I need to hide some fields based on login user's permissions. Whatever I did, I get an error when I want to get the id of the logged in user. It works when I give the id information myself (example: user with id 1). How can I solve this problem?

![screen-shot-1](https://user-images.githubusercontent.com/65877282/175832770-8a8f14e9-64af-481a-937f-917c5ad697b1.png)
![screen-shot-1-1](https://user-images.githubusercontent.com/65877282/175832769-b659000d-f9b6-48a9-9579-ce7dc1698904.png)
![screen-shot-2](https://user-images.githubusercontent.com/65877282/175832772-7d5c5eef-3698-4db9-a5e9-f7bf6f0fe1cd.png)
![screen-shot-3](https://user-images.githubusercontent.com/65877282/175832774-4cb302cb-b0af-4e39-8e59-d2a534f0fad8.png)

Another way:

![Screen Shot 2022-06-27 at 12 57 26](https://user-images.githubusercontent.com/65877282/175915628-a516ea47-467d-4153-b764-6fcfc88ff70e.png)
![Screen Shot 2022-06-27 at 12 58 10](https://user-images.githubusercontent.com/65877282/175915675-2b03b053-9ee6-451f-8092-22c369ccaf04.png)
![Screen Shot 2022-06-27 at 12 58 32](https://user-images.githubusercontent.com/65877282/175915749-be19feec-150f-4ba2-8cb2-69ca97b801a3.png)

Here 

User::with('roles')->where('id',**Auth::id()**)->first(); in its place
User::with('roles')->where('id',**1**)->first(); It works normally without any problems.

**Comment by oliveiraxavier** [comment]
Try as below
```
$userId = Auth::id();
return DataTables::of($users)
	->addColumn('action', function($user) use($userId) {
		$auth_user = User::with('roles')->where('id', $userId)->first();

	});
	/**others */
```
